pkg/cluster/spec/instance.go

```diff
@@ -164,6 +165,42 @@ func (i *BaseInstance) InitConfig(e executor.Executor, opt GlobalOptions, user s
 	return nil
 }
 
+// TransferLocalConfigFile scp local config file to remote
+// Precondition: the user on remote have permission to access & mkdir of dest files
+func (i *BaseInstance) TransferLocalConfigFile(e executor.Executor, local, remote string) error {
+	remoteDir := filepath.Dir(remote)
+	// make sure the directory exists
+	cmd := fmt.Sprintf("mkdir -p %s", remoteDir)
+	if _, _, err := e.Execute(cmd, false); err != nil {
+		return errors.Annotatef(err, "execute: %s", cmd)
+	}
+
+	if err := e.Transfer(local, remote, false); err != nil {
+		return errors.Annotatef(err, "transfer from %s to %s failed", local, remote)
+	}
+
+	return nil
+}
+
+// TransferLocalConfigDir scp local config directory to remote
+// Precondition: the user on remote have right to access & mkdir of dest files
+func (i *BaseInstance) TransferLocalConfigDir(e executor.Executor, local, remote string) error {
+	files, err := ioutil.ReadDir(local)
+	if err != nil {
+		return errors.Annotatef(err, "read local directory %s failed", local)
+	}
+
+	for _, f := range files {
```

Comment:
Do we need to restrict the suffix of the filename? like: `*.yml *.json`?

Reply:
I think so... We it's better to do that